### PR TITLE
Wine detail react query

### PR DESCRIPTION
**Pull request by AnnekeSchuette**
I created a Detail page for Wines using react-query and a custom hook, which will be extended by personal notes and ratings later on. Also I extracted the WineRecommendations into a hook.

Due to an update of prettier config and some CSS cleanup there are a lot of files with no or minor changes (line-breaks, styling, css variable names). You should concentrate on the non-css parts. 

Testing will be come with a separate PR to keep the scope small. Thanks in advance for your feedback and valuable comments ! 👩🏻‍🦰✌🏻

For Details see: https://github.com/AnnekeSchuette/capstone-project/issues/20

### CODE REVIEW

File: client/src/components/WineDetailPage/WineDetailPage.jsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  price: PropTypes.string,
+  averageRating: PropTypes.number,
+  ratingCount: PropTypes.number,
+  score: PropTypes.number,
+  link: PropTypes.string,
+}
+
+export default function WineDetailPage(...props) {
+  const { wineId } = useParams()
+  const [isLoading, error, currentWineData, isFetching] = useWineDetail(wineId)
```
Comment (marcushohlbein) [inline client/src/components/WineDetailPage/WineDetailPage.jsx:19]: really smart to put it inside a custom hook 😮

File: client/src/components/WineDetailPage/WineDetailPage.jsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  if (isLoading) {
+    return 'Is loading ...'
+  } else if (isFetching) {
+    return 'Updating ...'
+  } else if (error || currentWineData?.error) {
+    return `Oops, this should't happen ... 😬 ${
+      currentWineData.error.message === undefined
+        ? 'Wine not found'
+        : 'Error: ' + currentWineData.error.message
+    }`
```
Comment (marcushohlbein) [inline client/src/components/WineDetailPage/WineDetailPage.jsx:40]: good error handling 👍

File: client/src/hooks/useWineDetail.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+export default function useWineDetail(wineId) {
+  const {
+    isLoading,
+    error,
+    data: currentWineData,
+    isFetching,
+  } = useQuery('currentWineData', () => getWinebyId(wineId))
+
+  return [isLoading, error, currentWineData, isFetching]
+}
```
Comment (marcushohlbein) [inline client/src/hooks/useWineDetail.js:13]: cool, like it

File: client/src/index.js [hunk 4]
```diff
@@ -1,16 +1,29 @@
+import {
+  useQuery,
+  useMutation,
```
Comment (marcushohlbein) [inline client/src/index.js:3]: do you need to import this. didn't found where you use it?

File: client/src/index.js [hunk 5]
```diff
@@ -1,16 +1,29 @@
...
+} from 'react-query'
+import { ReactQueryDevtools } from 'react-query/devtools'
 import React from 'react'
 import ReactDOM from 'react-dom'
 import GlobalFonts from './fonts/Fonts'
 import App from 'components/App/App'
 import GlobalStyle from 'components/GlobalStyle'
 import { BrowserRouter as Router } from 'react-router-dom'
 
+const queryClient = new QueryClient()
```
Comment (marcushohlbein) [inline client/src/index.js:16]: you can add some additional config in here, like cachetime etc. .... if needed of course 🤗

**Review by marcushohlbein — APPROVED** [review]
very well implementation, added some comments in you code

File: client/src/index.js (hunk 4 above)
Comment (AnnekeSchuette) [reply client/src/index.js:3]: You're right! I will remove this import for now (maybe I need it in my next User Story when saving and editing notes).

File: client/src/index.js (hunk 5 above)
Comment (AnnekeSchuette) [reply client/src/index.js:16]: Good idea 👍🏻

**Pull request merged**